permissions-policy/resources/orientation-event.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+async function waitForOrientationEvent(eventName) {
+  if (eventName !== 'devicemotion' && eventName !== 'deviceorientation' && eventName !== 'deviceorientationabsolute') {
+    return 'ERROR';
+  }
+
+  let value;
+
+  try {
+    // See https://github.com/w3c/deviceorientation/issues/148: the
+    // specification currently does not fire any events when the permissions
+    // policy checks fail, so what we do here is wait for the timeout and the
+    // devicemotion event handler to race each other.
+    value = await new Promise((resolve, reject) => {
+      const timeoutId = window.setTimeout(() => {
+        window.removeEventListener(eventName, handler);
+        reject('NO-EVENT');
```

Comment:
Reject with real error if we do this... otherwise we lose the error stack.

Reply:
Is not having an error stack really a problem in this case? `test_feature_availability_with_post_message_result()` expects a string, so throwing an Error here would essentially require some plumbing that would ultimately result in a string comparison anyway.